src/components/tools/windowing/WindowLevelControls.vue

```diff
@@ -1,47 +1,231 @@
 <script lang="ts">
-import { defineComponent } from 'vue';
-import ToolButton from '@/src/components/ToolButton.vue';
+import { computed, defineComponent, ref } from 'vue';
 import { useCurrentImage } from '@/src/composables/useCurrentImage';
-import useWindowingStore from '@/src/store/view-configs/windowing';
+import useWindowingStore, {
+  defaultWindowLevelConfig,
+} from '@/src/store/view-configs/windowing';
 import { useViewStore } from '@/src/store/views';
+import { WLAutoRanges, WLPresetsCT, WL_AUTO_DEFAULT } from '@/src/constants';
+import { useDICOMStore } from '@/src/store/datasets-dicom';
 
 export default defineComponent({
-  components: {
-    ToolButton,
-  },
   setup() {
     const { currentImageID } = useCurrentImage();
     const windowingStore = useWindowingStore();
     const viewStore = useViewStore();
+    const dicomStore = useDICOMStore();
+    const panel = ref([1]);
+    const windowingDefaults = defaultWindowLevelConfig();
 
-    const resetWindowLevel = () => {
+    // Get the relevant view ids
+    const viewIDs = computed(() =>
+      viewStore.viewIDs.filter(
+        (viewID) => !!windowingStore.getConfig(viewID, currentImageID.value)
+      )
+    );
+
+    function parseLabel(text: string) {
+      return text.replace(/([A-Z])/g, ' $1').trim();
+    }
+
+    // --- Automatic Range Options --- //
+
+    const filteredWLAutoRanges = computed(
+      () =>
+        Object.fromEntries(
+          Object.entries(WLAutoRanges).filter(([, value]) => value !== 0)
+        ) as Record<string, number>
+    );
+
+    // --- CT Preset Options --- //
+
+    const modality = computed(() => {
+      if (
+        currentImageID.value &&
+        currentImageID.value in dicomStore.imageIDToVolumeKey
+      ) {
+        const volKey = dicomStore.imageIDToVolumeKey[currentImageID.value];
+        const { Modality } = dicomStore.volumeInfo[volKey];
+        return Modality;
+      }
+      return '';
+    });
+    const isCT = computed(
+      () =>
+        modality.value &&
+        ['ct', 'ctprotocol'].includes(modality.value.toLowerCase())
```

Comment:
This array should be made into a named constant for better visibility in the event it needs updating.

Reply:
Fixed.